Manual Pages from Sphinx (#1972)
longturn/freeciv21: docs/Manuals/Program/freeciv21-server.rst

website at https://longturn.readthedocs.io/.

This is the server `program` used to establish a Freeciv21 server. For more information on the game, refer
to freeciv21-client(6).

This manual page only lists the command-line arguments for the server `program`. For details of the
directives a server administrator can use to control aspects of the server, see freeciv21-game-manual(6).

OPTIONS
=======

The following options are accepted on the command-line of the server. They may not be combined as with other
tools. For example: ``freeciv21-server -fp savegame.sav 5556`` will not work. Instead you will need to enter
each option separately, such as, ``freevic21-server -s savegame.sav -p 5556``.

.. include:: /Manuals/Server/command-line.rst
    :start-line: 15

EXAMPLES
========

Here are some examples:

freeciv21-server --file oldgame.sav -p 2224
  Starts a server on port `2224`, loading the savegame named `oldgame.sav`.

freeciv21-server -R ranklog -l logfile -r start_script -f oldgame.sav.xz -p 2224
  Starts a server on port `2224`, loading the savegame named `oldgame.sav.xz`. Player ranking events are

>   logged to `ranking`, other logging events are placed in `logfile`. When the server first starts, it

Reviewer: Should `ranking` say `ranklog` instead in this sentence? The command is using `ranklog`

  immediately executes the server settings script named `start_script`. See freeciv21-server-manual(6) for
  more information on the server settings script.

> freeciv21-server -a -D users.conf -G -N -q 60  -l logfile -s ./saves -p 2224

Reviewer: ```suggestion
freeciv21-server -a -D users.conf -G -N -q 60 -l logfile -s ./saves -p 2224
```

  Starts a server on port `2224` with authentication enabled (`-a`, `-D`, `-G`, and `-N`). If no activity
  for 60 seconds the server will quit (`-q`). Logging events are placed in `logfile` and periodic saves are
  placed in the `./saves` directory.

> FILES

Reviewer 2: We should mention `database.lua` in this section.

=====

> The freeciv21-server requires the default ruleset files to be readable in the Freeciv21 `data` directory.

Reviewer 2: I found this a bit hard to understand. I would prefer something more generic.

> This is most often `/usr/local/share/freeciv21`.

Reviewer 2: Is this true for our debs? It's certainly not for snaps

> * civ2civ3/buldings.ruleset

Reviewer 2: ```suggestion
* civ2civ3/buildings.ruleset
```

* civ2civ3/cities.ruleset
* civ2civ3/effects.ruleset
* civ2civ3/game.ruleset
* civ2civ3/governments.ruleset
* civ2civ3/nations.rulest
* civ2civ3/styles.ruleset
* civ2civ3/techs.ruleset
* civ2civ3/terrain.ruleset
* civ2civ3/units.ruleset
* civ2civ3/script.lua
* default/default.lua
* default/nationlist.ruleset

These files are for the default ruleset for the game (civ2civ3), which are loaded if you do not use the

> `--ruleset` arguement. Alternate rules can be loaded with the ``rulsetdir`` directive in a start up script.
> Type ``help resetdir`` at the server command prompt for more information.

Reviewer 2: ```suggestion
`--ruleset` arguement. Alternate rules can be loaded with the ``rulesetdir`` directive in a start up script.
Type ``help rulesetdir`` at the server command prompt for more information.
```

ENVIRONMENT
===========

The Freeciv21 server accepts these environment variables:

FREECIV_CAPS
  A string containing a list of "capabilities" provided by the server. The compiled default should be
  correct for most purposes, but if you are familiar with the capability string facility in the source you
  may use it to enforce some constraints between clients and server.

FREECIV_COMPRESSION_LEVEL
  Sets the compression level for network traffic.

FREECIV_DATA_ENCODING
  Sets the character encoding used for data files, savegames, and network strings. This should not normally
  be changed from the default of UTF-8, since that is the format of the supplied rulesets and the standard
  network protocol.

FREECIV_INTERNAL_ENCODING
  Sets the character encoding used internally by freeciv21-server. This encoding should not be visible at
  any interface. Defaults to UTF-8.

FREECIV_LOCAL_ENCODING
  Sets the local character encoding (used for the command line and terminal output). The default is inferred

>   from other aspects of the environment.

Reviewer 2: Don't advertise them. The data and internal encoding should always be UTF-8. The local encoding should be determined by Qt.

  looks in the following directories, in order, for any data files: the current directory; the "data"
  subdirectory of the current directory; the subdirectory ".local/share/freeciv21" in the user's home
  directory; and the directory where the files are placed by running "cmake --target install".

FREECIV_SAVE_PATH
  A colon separated list of directories pointing to the Freeciv21 save directories. By default Freeciv21
  looks in the following directories, in order, for save files: the current directory; and the subdirectory
  ".local/share/freeciv21/saves" in the user's home directory.

  (This does not affect where the server creates save game files. See the `--saves` option for that.)

FREECIV_SCENARIO_PATH
  A colon separated list of directories pointing to the Freeciv21 scenario directories. By default Freeciv21
  looks in the following directories, in order, for scenario files: the current directory; the
  "data/scenarios" subdirectory of the current directory; the subdirectory
  ".local/share/freeciv21/scenarios" in the user's home directory; and the directory where the files are
  placed by running "cmake --target install".

>   (This does not affect where the server creates scenario files See the `--scenarios` option for that.)

Reviewer 2: ```suggestion
  (This does not affect where the server creates scenario files. See the `--scenarios` option for that.)
```
